packages/flutter_tools/test/commands.shard/hermetic/host_validator_test.dart

```diff
@@ -0,0 +1,84 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter_tools/src/doctor.dart';
+import 'package:flutter_tools/src/host_validator.dart';
+import 'package:flutter_tools/src/base/platform.dart';
+import 'package:http/http.dart';
+import 'package:http/testing.dart';
+
+import '../../src/common.dart';
+
+void main() {
+  group('host validator', () {
+    const List<String> osTested = <String>['windows', 'macos', 'linux'];
+    final Future<ValidationResult> Function(Client mockClient, String os) runHostTest
```

Comment:
Ditto with [style guide](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#formatting) and formatting (for example, blocks should be indented 2 columns, it looks like these are indented 4).

Reply:
Also, with these types of tests, rather than testing all three platforms in a single test, we usually nest the test itself within a loop, interpolating the platform into the testname. this way if there's a platform specific error, it's immediately obvious which platform has the issue. For example https://github.com/flutter/flutter/blob/master/packages/flutter_tools/test/general.shard/ios/devices_test.dart#L169